module/apmawssdkgo/sqs.go

```diff
@@ -0,0 +1,130 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package apmawssdkgo // import "go.elastic.co/apm/module/apmawssdkgo"
+
+import (
+	"errors"
+	"strings"
+
+	"go.elastic.co/apm"
+	"go.elastic.co/apm/module/apmhttp"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/sqs"
+)
+
+var (
+	errSQSMethodNotSupported = errors.New("method not supported")
+	operationName            = map[string]string{
+		"SendMessage":        "send",
+		"SendMessageBatch":   "send_batch",
+		"DeleteMessage":      "delete",
+		"DeleteMessageBatch": "delete_batch",
+		"ReceiveMessage":     "poll",
+	}
+)
+
+type apmSQS struct {
+	name, opName, resourceName string
+}
+
+func newSQS(req *request.Request) (*apmSQS, error) {
+	opName, ok := operationName[req.Operation.Name]
+	if !ok {
+		return nil, errSQSMethodNotSupported
+	}
+	name := req.ClientInfo.ServiceID + " " + strings.ToUpper(opName)
+	resourceName := serviceSQS
+
+	queueName := getQueueName(req)
+	if queueName != "" {
+		name += " " + operationDirection(req.Operation.Name) + " " + queueName
+		resourceName += "/" + queueName
+	}
+
+	s := &apmSQS{
+		name:         name,
+		opName:       opName,
+		resourceName: resourceName,
+	}
+
+	return s, nil
+}
+
+func (s *apmSQS) spanName() string { return s.name }
+
+func (s *apmSQS) resource() string { return s.resourceName }
+
+func (s *apmSQS) setAdditional(span *apm.Span) {
+	span.Action = s.opName
+	// TODO(stn): record `context.message.queue.name`
+}
+
+// addMessageAttributes adds message attributes to `SendMessage` and
+// `SendMessageBatch` RPC calls. Other SQS RPC calls are ignored.
+func addMessageAttributes(req *request.Request, span *apm.Span) {
+	switch req.Operation.Name {
+	case "SendMessage", "SendMessageBatch":
+		break
+	default:
+		return
+
+	}
+
+	traceContext := span.TraceContext()
+	msgAttr := &sqs.MessageAttributeValue{
+		DataType:    aws.String("String"),
+		StringValue: aws.String(apmhttp.FormatTraceparentHeader(traceContext)),
+	}
+
+	if req.Operation.Name == "SendMessage" {
+		input, ok := req.Params.(*sqs.SendMessageInput)
+		if !ok {
+			return
+		}
+		input.MessageAttributes["traceContext"] = msgAttr
```

Comment:
```suggestion
		input.MessageAttributes["Traceparent"] = msgAttr
```

there should be 1-3 attributes:
 - Traceparent (required)
 - Tracestate (optional, but should almost always be set)
 - Elastic-Apm-Traceparent (optional; depends on the result of the transaction's `ShouldPropagateLegacyHeader()` method)

The code should end up looking something like https://github.com/elastic/apm-agent-go/blob/0f4fcadbeb351a68c518bd88518f9903668973dd/module/apmhttp/client.go#L136-L145

For reference, here's the Ruby agent's test spec:
https://github.com/elastic/apm-agent-ruby/blob/4a3155ecebb7a189388f734d6a7478a05e80bc2f/spec/elastic_apm/spies/sqs_spec.rb#L68-L78